feat(curriculum): add cipher project to Python curriculum (#52373)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/6551fe3b1df7c9740f13f270.md

---
id: 6551fe3b1df7c9740f13f270
title: Step 11
challengeType: 20
dashedName: step-11
---

# --description--

> Before we go on, here are some key aspects about variable naming in Python you should keep in mind:

Reviewer: ```suggestion
Before we go on, here are some key aspects of variable naming in Python you should keep in mind:
```

- Some words like `for`, `while`, `True` and others are reserved keywords. They have a special meaning in Python and you cannot use them for naming your variables.
- Variable names cannot start with a number and can only contain alpha-numeric characters and underscores.

> - Variable names are case sentitive. Thats is `my_var` is different from `my_Var` or `MY_VAR`.

Reviewer: ```suggestion
- Variable names are case sensitive. Thats means, `my_var` is different from `my_Var` or `MY_VAR`.
```

- Finally, in Python it is a common convention to write variable names using `snake_case`, with each space replaced by an underscore character and the words written in lowercase letters.

> Now remove both your `print` calls, then create another variable called `alphabet` and assign it the value of the string `abcdefghijklmnopqrstuvwxyz`.

Reviewer: ```suggestion
Now remove both your `print` calls, then create another variable called `alphabet` and assign it the string value of `abcdefghijklmnopqrstuvwxyz`.
```

---
id: 6551fe3b1df7c9740f13f270
title: Step 11
challengeType: 20
dashedName: step-11
---

# --description--

> Before we go on, here are some key aspects of variable naming in Python you should keep in mind:
> 
> - Some words like `for`, `while`, `True` and others are reserved keywords. They have a special meaning in Python and you cannot use them for naming your variables.
> - Variable names cannot start with a number and can only contain alpha-numeric characters and underscores.
> - Variable names are case sensitive. That means `my_var` is different from `my_Var` or `MY_VAR`.
> - Finally, in Python it is a common convention to write variable names using `snake_case`, with each space replaced by an underscore character and the words written in lowercase letters.
> 
> Now remove both your `print` calls, then create another variable called `alphabet` and assign it the string value of `abcdefghijklmnopqrstuvwxyz`.

Reviewer 2: ```suggestion
Key aspects of variable naming in Python are:

- Some words are reserved keywords (e.g. `for`, `while`, `True`). They have a special meaning in Python, so you cannot use them for variable names.
- Variable names cannot start with a number, and can only contain alpha-numeric characters or underscores.
- Variable names are case sensitive, i.e. `my_var` is different to `my_Var` and `MY_VAR`.
- Finally, it is a common convention to write variable names using `snake_case`, where each space is replaced by an underscore character and the words written in lowercase letters.

Remove both your `print` calls, then create another variable called `alphabet`, and assign it the string value of `abcdefghijklmnopqrstuvwxyz`.
```
